src/code/templates/zoom-challenge.js

```diff
@@ -6,10 +6,47 @@ export default class ZoomChallenge extends Component {
 
   static backgroundClasses = 'fv-layout fv-layout-zoom'
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasTutorial: false
+    };
+  }
+
   componentDidMount() {
     var iframeElement = document.getElementById("iframe");
     phone = new iframePhone.ParentEndpoint(iframeElement);
-    phone.addListener('challengeWin', this.props.onWinZoomChallenge);
+    phone.addListener('challengeWin', (activityData) => {
+      let stats;
+      const positiveMetrics = this.props.zoomScoringMetrics;
+
+      // check we have a list of positive metrics and score to calculate
+      if (activityData && activityData.actionStats && positiveMetrics) {
+        stats = activityData.actionStats;
+
+        if (positiveMetrics && stats) {
+          let total = 0, positiveTotal = 0;
+          for (let m in stats) {
+            if (positiveMetrics.indexOf(m) > -1) {
+              positiveTotal += stats[m];
```

Comment:
As we discussed in our zoom meeting, if the metrics were a set of weights rather than a list of strings, then `positiveTotal += stats[m] * (weights[m] || 0)` and `total += stats[m] * Math.abs(weights[m] || 0)` would be a bit more flexible.